tasks/deploy/sidechain.ts

```diff
@@ -79,25 +79,29 @@ task("deploy:sidechain:L1")
         const phase2 = await config.getPhase2(deployer);
         const phase6 = await config.getPhase6(deployer);
 
-        const result = await deployCanonicalPhase(hre, config.addresses, phase2, phase6, deployer, debug, tskArgs.wait);
+        const result = await deployCanonicalPhase(hre, deployer, config.addresses, phase2, phase6, debug, tskArgs.wait);
         logContracts(result as unknown as { [key: string]: { address: string } });
     });
 
 task("deploy:sidechain:L2")
     .addParam("wait", "wait for blocks")
     .setAction(async (tskArgs: TaskArguments, hre: HardhatRuntimeEnvironment) => {
         const deployer = await getSigner(hre);
+        const l1Config = canonicalConfigs[hre.network.config.chainId];
```

Comment:
This is confusing, you are using the `hre.network.config.chainId` and looking it up on the canonical configs object but this is the L2 deployment task 🤔

Reply:
I removed any dependency on l1 for this task